### PR TITLE
[IMP] crm: use only one field for leads/opportunities count

### PR DESCRIPTION
**Pull request by qmo-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/crm/models/utm.py [hunk 1]
```diff
@@ -6,37 +6,25 @@
 class UtmCampaign(models.Model):
     _inherit = 'utm.campaign'
 
-    crm_lead_activated = fields.Boolean('Use Leads', compute='_compute_crm_lead_activated')
-    lead_count = fields.Integer('Lead Count', groups='sales_team.group_sale_salesman', compute="_compute_global_opportunity_and_lead_count")
-    opportunity_count = fields.Integer('Opportunity Count', groups='sales_team.group_sale_salesman', compute="_compute_global_opportunity_and_lead_count")
+    use_leads = fields.Boolean('Use Leads', compute='_compute_use_leads')
+    leads_opportunities_count = fields.Integer('Leads/Opportunities count', groups='sales_team.group_sale_salesman', compute="_compute_leads_opportunities_count")
```
Comment (tde-banana-odoo) [inline addons/crm/models/utm.py]: I propose to name it crm_lead_count to have shorter name (and more coherent with model itself). What do you think ?

File: addons/crm/models/utm.py [hunk 2]
```diff
@@ -6,37 +6,25 @@
...
-            for campaign in self:
-                campaign.lead_count = 0
-                campaign.opportunity_count = data_map.get(campaign.id, 0)
-
-    def action_redirect_to_leads(self):
-        action = self.env.ref('crm.crm_lead_all_leads').read()[0]
-        action['domain'] = [('campaign_id', '=', self.id)]
-        action['context'] = {'default_type': 'lead', 'active_test': False}
-        return action
+        mapped_data = {datum['campaign_id'][0]: datum['campaign_id_count'] for datum in lead_data}
```
Comment (tde-banana-odoo) [inline addons/crm/models/utm.py]: Still out old datum :heart:

File: addons/crm/models/utm.py [hunk 3]
```diff
@@ -6,37 +6,25 @@
...
-    def action_redirect_to_opportunities(self):
-        action = self.env.ref('crm.crm_lead_opportunities').read()[0]
+    def action_redirect_to_leads_opportunities(self):
+        view = 'crm.crm_lead_all_leads' if self.use_leads else 'crm.crm_lead_opportunities'
+        action = self.env.ref(view).read()[0]
         action['view_mode'] = 'tree,kanban,graph,pivot,form,calendar'
-        action['domain'] = [('campaign_id', '=', self.id)]
-        action['context'] = {'active_test': False}
+        action['domain'] = [('campaign_id', 'in', self.ids)]
+        action['context'] = {'active_test': False, 'create': False}
```
Comment (tde-banana-odoo) [inline addons/crm/models/utm.py]: Why specifically create False here ?

File: addons/mass_mailing_crm/models/mailing_mailing.py [hunk 4]
```diff
@@ -10,50 +10,26 @@ class MassMailing(models.Model):
     _name = 'mailing.mailing'
     _inherit = 'mailing.mailing'
 
-    crm_lead_activated = fields.Boolean('Use Leads', compute='_compute_crm_lead_activated')
-    crm_lead_count = fields.Integer('Lead Count', groups='sales_team.group_sale_salesman', compute='_compute_crm_lead_and_opportunities_count')
-    crm_opportunities_count = fields.Integer('Opportunities Count', groups='sales_team.group_sale_salesman', compute='_compute_crm_lead_and_opportunities_count')
+    use_leads = fields.Boolean('Use Leads', compute='_compute_use_leads')
+    leads_opportunities_count = fields.Integer('Leads/Opportunities Count', groups='sales_team.group_sale_salesman', compute='_compute_leads_opportunities_count')
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_crm/models/mailing_mailing.py]: Same remark here about naming, if we change it just propagate ^^ .

File: addons/crm/views/utm_campaign_views.xml [hunk 5]
```diff
@@ -5,19 +5,15 @@
...
-                <div class="mr-3"
-                    groups="sales_team.group_sale_salesman"
-                    attrs="{'invisible': [('crm_lead_activated', '=', False)]}" title="Leads">
-                    <i class="fa fa-star text-muted"></i>
-                    <small class="font-weight-bold"><field name="lead_count"/></small>
-                </div>
                 <div class="mr-3"
                     groups="sales_team.group_sale_salesman"
-                    attrs="{'invisible': [('crm_lead_activated', '=', True)]}" title="Opportunities">
+                    t-att-title="record.use_leads.raw_value ? 'Leads' : 'Opportunities'">
```
Comment (tde-banana-odoo) [inline addons/crm/views/utm_campaign_views.xml]: It is not really correct as both are computed in the read group it seems ... Did you have a PO final word on it ? Like Opportunities or Leads & Opportunities maybe ?

File: addons/crm/models/utm.py (hunk 3 above)
Comment (qmo-odoo) [reply addons/crm/models/utm.py]: At first it was a copy/paste mistake but it ended up being an LNA approved change. Life is indeed like a chocolate box after all

File: addons/crm/views/utm_campaign_views.xml (hunk 5 above)
Comment (qmo-odoo) [reply addons/crm/views/utm_campaign_views.xml]: On that front, I kept the same behavior as before which has been re-approved by LNA. Maybe one day we'll decide to actually use two fields with correct values by using the flag but as of now they do not want that. Also, if we changed it there, we would also have to change it in the stat buttons of the forms which would just not work due to the length of the label.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**